docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,387 @@
+---
+approvers:
+- mikedanese
+- luxas
+- errordeveloper
+- jbeda
+title: Kubeadm init
+notitle: true
+---
+{% capture overview %}
+## Kubeadm init {#cmd-init}
+{% endcapture %}
+
+{% capture body %}
+{% include_relative _kubeadm/kubeadm_init.md %}
+
+### Init workflow {#init-workflow}
+`kubeadm init` bootstraps a Kubernetes master node by executing the
+following steps:
+
+1. kubeadm runs a series of pre-flight checks to validate the system state
+   before making changes. Some checks only trigger warnings, others are
+   considered errors and will exit kubeadm until the problem is corrected or the
+   user specifies `--skip-preflight-checks`.
+
+1. kubeadm generates a token that additional nodes can use to register
+   themselves with the master in future.  Optionally, the user can provide a
+   token via `--token`, as described in the
+   [kubeadm token](kubeadm-token.md).
+
+1. kubeadm generates a self-signed CA to provision identities for each component
+   (including nodes) in the cluster.  It also generates client certificates to
+   be used by various components.  If the user has provided their own CA by
+   dropping it in the cert directory configured via `--cert-dir`
+   (`/etc/kubernetes/pki` by default) this step is skipped as described in the
+   [Using custom certificates](#custom-certificates) document.
+
+1. kubeadm writes kubeconfig files in `/etc/kubernetes/`  for
+   the kubelet, the controller-manager and the scheduler to use to connect to the
+   API server, each one with their respective identities, as well as an additional
+   kubeconfig file for administration.
+
+1. kubeadm generates static Pod manifests for the API server,
+   controller manager and scheduler; in case an external etcd is not provided,
+   an additional static Pod manifest will be generated for etcd.
+
+   Static Pod manifests are written in `/etc/kubernetes/manifests`; the kubelet
+   watches this directory for Pods to create on startup.
+
+   Once control plane Pods are up and running kubeadm init sequence can continue.
+
+1. kubeadm "labels" and "taints" the master node so that only control plane
+   components will run there.
+
+1. kubeadm makes all the necessary configurations for allowing node joining with the
+   [Bootstrap Tokens](/docs/admin/bootstrap-tokens/) and
+   [TLS Bootstrap](/docs/admin/kubelet-tls-bootstrapping/)
+   mechanism:
+
+   - Write a ConfigMap for making available all the information required
+     for joining and set up related RBAC access rules.
+
+   - Ensure access to the CSR signing API for bootstrap tokens.
+
+   - Configure auto approval for new CSR requests.
+
+   See [kubeadm join](kubeadm-join.md) for additional info.
+
+1. kubeadm installs add-on components via the API server.  Right now this is
```

Comment:
Maybe it's worth adding a note here that although the DNS server is deployed, it will not be scheduled until CNI is installed.